griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,270 @@
+                return ErrorArtifact(f"error appending to Google Doc, file not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error appending text to Google Doc with path {file_path}: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to prepend text to a Google Doc",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path on Google Drive in the POSIX format. "
+                                    "For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be prepended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def prepend_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "v1", "docs")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+
+                if len(doc["body"]["content"]) == 1:
+                    requests = [{"insertText": {"location": {"index": 1}, "text": text}}]
+                else:
+                    start_index = doc["body"]["content"][1]["startIndex"]
+                    requests = [{"insertText": {"location": {"index": start_index}, "text": text}}]
+
+                doc = service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("text prepended successfully")
+            else:
+                return ErrorArtifact(f"error prepending to google doc, file not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error prepending text to Google Doc with path {file_path}: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to create a new Google Doc and save content to it on Google Drive",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Name of the file to be created, which will be used to save content in."
+                    ): str,
+                    Literal(
+                        "content",
+                        description="Content to be saved in Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def save_content_to_google_doc(self, params: dict) -> ErrorArtifact | InfoArtifact:
```

Comment:
Why have two separate activities to create and save content to a Google Doc? We can use `save_content_to_google_doc` to achieve both, making `content` optional.

Reply:
We should also have another activity that loads content from memory.

Reply:
merged create and save

Reply:
made save_memory_artifacts_to_google_doc